src/basic/copy.c

```diff
@@ -114,9 +112,9 @@ int copy_bytes(int fdf, int fdt, uint64_t max_bytes, bool try_reflink) {
 
                 /* As a fallback just copy bits by hand */
                 {
-                        uint8_t buf[m];
+                        uint8_t buf[MIN(m, COPY_BUFFER_SIZE)];
```

Comment:
LLVM/clang will complain about VLAs being a C99 feature, but then again, this was a VLA before, so that's not a regression.